app/controllers/orangelight/browsables_controller.rb

```diff
@@ -1,59 +1,179 @@
 # frozen_string_literal: true
 
 class Orangelight::BrowsablesController < ApplicationController
-  # GET /orangelight/names
-  # GET /orangelight/names.json
-  def index
+  def update_facet
+    if browsing_names?
+      @facet = 'author_s'
+    elsif browsing_subjects?
+      @facet = 'subject_facet'
+    elsif browsing_titles?
+      @facet = 'name_title_browse_s'
+    end
+  end
+
+  # All of this needs to be restructured into a QueryService Class
+  def connection
+    Orangelight.browse_lists_index.connection
+  end
+
+  def odm_class
+    return CallNumberDocument if browsing_call_numbers?
+
+    BrowseListDocument
+  end
+
+  def query_response_documents(query_response)
+    query_response["response"]["docs"].map { |doc| odm_class.new(doc) }
+  end
+
+  ## The queries
+
+  def solr_default_params
+    {
+      sort: solr_default_sort,
+      fl: solr_default_fields,
+      rows: 100_000_0
```

Comment:
I was uncertain how to locate the constant for this. From what I understoo, Ruby does not really offer a solid ceiling for `Integer`.